src/problem.jl

```diff
@@ -0,0 +1,147 @@
+abstract type AbstractProblem end
+
+"""
+    compute(problem, grid::Union{Grid,GridRSP})
+
+Compute problem `o` on a grid.
+"""
+function compute end
+
+"""
+    assess(p::AbstractProblem, g)
+
+Assess the memory and compute requirements of problem
+`p` on grid `g`. This can be used to indicate memory
+and time reequiremtents on a cluster
+"""
+function assess end
+
+"""
+    SolverMode
+
+Abstract supertype for ConScape solver modes.
+"""
+abstract type SolverMode end
+"""
+   MaterializedSolve()
+
+Compute all operations on a fully materialised Z matrix.
+
+This is inneficient for CPUS, but may be best for GPUs
+using CuSSP.jl ?
+"""
+struct MatrixSolve <: SolverMode end
+"""
+   ColumnSolve()
+
+Compute all operations column by column,
+after precomputing lu decompositions.
+
+TODO we can put LinearSolve.jl solver objects in the struct
```

Comment:
Yes, I definitely think we should do that!

Reply:
We also should put a LinearProblem in the struct, to be able to reuse the LU factors and other expensive computations that are cached by LinearProblem

Reply:
Oh, something else: I think that we should have a field that relates to landmarks. Landmarks relate to the problem, not to the grid. The fact that one needs to define two grids to obtain landmarks shows that the current implementation is not appropriate (see https://conscape.org/notebooks/nbk_landmarks.html)

Reply:
One problem with putting the LinearProblem directly in the struct is that it needs spatial inputs `A` and `b`. It would be nice to keep the `Problem` separate from spatial locations so we can reuse it over multiple windows. So I'm thinking more of holding the keywords for `LinearProblem` rather than the object itself

Reply:
That makes total sense.